### PR TITLE
feat: add support for hooks metadata details

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds processing for hook-related info, namely `EmitDetails`, `HookParameters`, and `HookExecutions` to the `Detailed` tab for transactions. It also fixes an issue where the Explorer thought hook-emitted transactions were sent with a ticket, due to an assumption that used to be true but is not with hooks.

### Context of Change

Better hooks support

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

### TypeScript/Hooks Update

All files touched already use Hooks and TypeScript.

### Sequence
Before:
image
image

After:
image
image

### Hook Details
image
image

## Test Plan

CI passes. Works locally.

### CODE REVIEW

**Review by dangell7 — APPROVED** [review]

File: src/containers/Accounts/AccountHeader/BalanceSelector/BalanceSelector.tsx [hunk 1]
```diff
@@ -28,6 +28,7 @@ export const BalanceSelector = ({
         return (
           <BalanceSelectorItem
             currency={currency}
+            key={currency}
```
Comment (ckniffen) [inline src/containers/Accounts/AccountHeader/BalanceSelector/BalanceSelector.tsx:31]: Woops.  I forgot to add this.

Comment (mvadari) [reply src/containers/Accounts/AccountHeader/BalanceSelector/BalanceSelector.tsx:31]: The console yelled at me every time I opened an account page, and it was a quick fix

**Review by pdp2121 — APPROVED** [review]

File: src/containers/shared/components/Sequence.tsx [hunk 2]
```diff
@@ -7,27 +7,33 @@ interface SequenceProps {
...
   ticketSequence = 0,
   account = '',
+  isHook = false,
 }) => {
   const { t } = useTranslation()
   const isPseudoTransaction = account === ACCOUNT_ZERO || account === ''
 
+  function getContext() {
+    if (isHook) {
+      return addContextHelp === true ? t('hook_emitted') : t('hook')
```
Comment (ckniffen) [inline src/containers/shared/components/Sequence.tsx:25]: Can you add a new test for this?

File: src/containers/Transactions/DetailTab.tsx [hunk 3]
```diff
@@ -117,11 +118,140 @@ export const DetailTab: FC<{ data: any }> = ({ data }) => {
       </div>
     ) : null
 
+  const renderHooks = () => {
```
Comment (ckniffen) [inline src/containers/Transactions/DetailTab.tsx:121]: This is large enough to be its own component.

File: src/containers/Transactions/DetailTab/HookDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+      {tx.EmitDetails && <EmitDetails emitDetails={tx.EmitDetails} />}
+      {tx.HookParameters && (
+        <div className="detail-subsection" data-test="hook-params">
+          <div className="detail-subtitle">Hook Parameters</div>
+          {tx.HookParameters.map(HookParameter)}
+        </div>
+      )}
+      {meta.HookExecutions && (
+        <div className="detail-subsection" data-test="hook-executions">
+          <div className="detail-subtitle">Hook Executions</div>
```
Comment (ckniffen) [inline src/containers/Transactions/DetailTab/HookDetails.tsx:120]: i18n this value.

File: src/containers/Transactions/DetailTab/HookDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+      />
+    </li>
+    <li className="detail-line">
+      <Trans
+        i18nKey="emit_parent"
+        values={{
+          hash: `${emitDetails.EmitParentTxnID.substring(20)}...`,
+        }}
+      >
+        <Link to={`/transactions/${emitDetails.EmitParentTxnID}`} />
```
Comment (ckniffen) [inline src/containers/Transactions/DetailTab/HookDetails.tsx:39]: Use new component `RouteLink` so that this will work with custom.

**Review by ckniffen — CHANGES_REQUESTED** [review]

**Pull request merged**